CMakeLists.txt

```diff
@@ -0,0 +1,39 @@
+project(facil.io C)
+
+find_package(Threads REQUIRED)
+
+set(facil.io_SOURCES
+  src/services/pubsub.c
+  src/http/http1_simple_parser.c
+  src/http/http_request.c
+  src/http/websockets.c
+  src/http/http1_request.c
+  src/http/http_response.c
+  src/http/unused/hpack.c
+  src/http/http.c
+  src/http/http1.c
+  src/http/http1_response.c
+  src/bscrypt/bscrypt/sha1.c
+  src/bscrypt/bscrypt/hex.c
+  src/bscrypt/bscrypt/misc.c
+  src/bscrypt/bscrypt/siphash.c
+  src/bscrypt/bscrypt/random.c
+  src/bscrypt/bscrypt/base64.c
+  src/bscrypt/bscrypt/unused/mempool.c
+  src/bscrypt/bscrypt/sha2.c
+  src/bscrypt/bscrypt/xor-crypt.c
+  src/core/defer.c
+  src/core/facil.c
+  src/core/sock.c
+  src/core/evio.c
+)
+
+add_library(facil.io ${facil.io_SOURCES})
+target_link_libraries(facil.io PRIVATE Threads::Threads)
+target_include_directories(facil.io
```

Comment:
> Would you consider having an include folder for applications that use facil.io?

I know it makes sense to have a separate `include` (vs. `src`) folder, but I find it confusing for development purposes... too often, the `include` folder becomes one long list of files that's very hard to navigate.

I prefer making having a [dynamic `makefile` implementation](https://github.com/boazsegev/facil.io/blob/b4680d11e72b7e4864ec1afc611de8cdbbcddd36/makefile#L52-L59) (or a dynamic CMake implementation) than making source-code navigation harder.

> Which headers will be needed by applications that use facil.io?

I guess that depends what each application does.

If the application uses the HTTP extensions, than it should include the `http.h` or the `websockets.h` files. However, these files have their own `include` directives for the `facil.h`, `http_request.h` and `http_response.h` files.

If the application uses pub/sub, the `pubsub.h` include file would also be required.

On the other hand, an "echo" application, or someone writing a custom protocol, could probably include just the `facil.h` file... which has `include` directives for `defer.h` and `sock.h`. These applications can also ignore the `http`, `services` and `bscrypt` folders or delete them, since the `core` is designed to be a stand-alone package.

The helper `fio_*` types are also great to have (they are used by the `core`, although there are other (perhaps better) alternatives to the `fio_list.h` and `fio_hash_table.h` implementations... since I favored security and ease of use over performance (i.e., the hash function implementation uses SipHash instead of CityHash or Murmur3).